packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -2429,19 +2443,27 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
    */
   selectData(columnIndex: ModelIndex, rowIndex: ModelIndex): void {
     const { model } = this.props;
-    const { columns } = model;
+    const { columns, groupedColumns } = model;
     const dataMap: Record<
       string,
-      { value: unknown; text: string | null; type: string }
+      {
+        value: unknown;
+        text: string | null;
+        type: string;
+        isExpandable: boolean;
+        isGrouped: boolean;
+      }
```

Comment:
We should define this type above, use it in `onDataSelected` and here.
Also, `text` should just be `string` - we don't allow `null` there.

We can't import the Linker type directly since IrisGrid doesn't depend on linker - either way there should be some abstraction between the two.